csharp/ql/src/semmle/code/csharp/Location.qll

```diff
@@ -171,6 +171,9 @@ class Assembly extends Location, Attributable, @assembly {
   /** Gets the version of this assembly. */
   Version getVersion() { assemblies(this, _, _, _, result) }
 
+  /** Gets an attribute attached to this assembly, if any. */
+  final Attribute getAnAttribute() { attributes(result, _, this) }
```

Comment:
`result = this.(Attributable).getAnAttribute()` ?

Reply:
Doesn't work, as `Assembly` is no longer a sub type of `Attributable`.

Reply:
Won't it cast via the db type?

Reply:
Nope, it will cast via `Attributable` which will restrict it to `Element`.